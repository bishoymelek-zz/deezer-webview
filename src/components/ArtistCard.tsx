import React from 'react';
import { Card } from 'antd';

const ArtistDetailsComponent = (props: any): JSX.Element => {
  const { trackListUrl } = props;
  return (
    <>
      <a target="_blank" rel="noopener noreferrer" href={trackListUrl}>
        Listen now
      </a>
    </>
  );
};
type ArtistCardPropType = {
  name: string;
  id: string;
  picture: string;
  trackListUrl: string;
};

export default function ArtistCard(props: ArtistCardPropType): JSX.Element {
  const { picture, name, trackListUrl } = props;
  return (
    <Card hoverable bordered cover={<img alt="artist-img" src={picture} />}>
      <Card.Meta
        title={name}
        description={<ArtistDetailsComponent trackListUrl={trackListUrl} />}
      />
    </Card>
  );
}
