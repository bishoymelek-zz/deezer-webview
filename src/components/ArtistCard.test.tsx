import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ArtistCard from './ArtistCard';

const mockStore = configureStore([]);

describe('Component/ArtistCard', () => {
  let store;
  let component;
  const artistCardMockData = {
    name: 'Hi',
    id: '1',
    picture: 'pic.png',
    trackListUrl: 'https://placeholder.com',
  };
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <ArtistCard
          name={artistCardMockData.name}
          id={artistCardMockData.id}
          picture={artistCardMockData.picture}
          trackListUrl={artistCardMockData.trackListUrl}
        />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
