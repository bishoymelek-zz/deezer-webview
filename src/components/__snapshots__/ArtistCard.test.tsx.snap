// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/ArtistCard should render with given state from Redux store 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable"
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="artist-img"
      src="pic.png"
    />
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-title"
        >
          Hi
        </div>
        <div
          className="ant-card-meta-description"
        >
          <a
            href="https://placeholder.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Listen now
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
